### DISCUSSION TITLE
[bug] GitHub mobile always shows auto-merge as REBASE even when it’s actually set to SQUASH

### DISCUSSION THREAD

**Discussion by TheRubenM** [community/community#65461] [Mobile]
My team uses the auto-merge feature quite a bit. When there are several commits, we opt to enable auto-merge with a SQUASH & MERGE setting. However, on the mobile app, it always shows it as set to REBASE and merge. This has led to lots of confusion and false alarms when you're double-checking your PR from a mobile device and think you did the wrong thing. The mobile screenshot attached is of a PR that is definitely set to SQUASH & MERGE on Desktop. 

![image](https://github.com/community/community/assets/13063893/754e1a3d-49f2-4c5c-8c10-9a0dc2682c7e)

<details>
<summary>ℹ️ Additional Debug Info</summary>
GitHub for iOS, Version 1.127.2 (108489730)
Device: iPhone 12 Pro Max (iOS 17.0)
</details>

**Comment by aconsuegra** [comment]
Hi @TheRubenM, thanks for reaching out and reporting this issue. This looks clearly like a serious bug that we need to solve, so I'll create an internal tracking issue for our engineers to look into it.

**Reply by aconsuegra** [reply]
A quick follow up @TheRubenM, does you team see this issue on both iOS and Android?

**Reply by aconsuegra** [reply]
We are currently looking into this issue @TheRubenM, so I have more questions that would help us debug it. 

- When you reproduce this issue, do you always enable Auto-Merge in github.com? Or also when you enable it on the Mobile apps?
- What are the merge options that you have enabled for your repository in settings? 

Thanks in advance

**Reply by TheRubenM** [reply]
Hi @aconsuegra , thanks for looking into it! Responses: 
- iOS v. Android: Unfortunately unable to check Android as everyone on my team is currently on iOS. 
- .com vs mobile app: I'd only tried setting auto-merge on github.com before, but I just tried setting auto-merge from the mobile app and it _seems_ to say squash and merge properly on the app when it's set from the app. 
- Attaching a screenshot of what (I think) you're asking re: merge settings. LMK if you were asking about something different though!
![image](https://github.com/community/community/assets/13063893/e0e201a3-044c-4a58-959b-16d73cdfb480)

**Reply by aconsuegra** [reply]
Thanks for the screenshot, this is exactly what I was looking for!

**Comment by aconsuegra** [comment]
@TheRubenM I think we have found the problem and we will ship a potential fix in next week's beta. That build should be available the following week (September 11th) in production.

If you are interested in validating the bug fix in Beta next week, we can share a link to join our TestFlight. Just let us know!

**Reply by TheRubenM** [reply]
Joining TestFlight would be awesome! Thank you!!

**Reply by aconsuegra** [reply]
Thanks a lot! You can join via this link https://testflight.apple.com/join/NLskzwi5

**Comment by jmeridth** [comment]
@aconsuegra This is still happening in the mobile (beta) latest version. My repo only allows `squash and merge` but mobile is defaulting to `rebase and merge`. Even after switching to `squash and merge` in the mobile app it errors about `rebase and merge`

**Reply by aconsuegra** [reply]
Hi @jmeridth, we are putting together a fix as soon as we can, it will hopefully be available today in a new app update.

**Reply by jmeridth** [reply]
Thanks @aconsuegra. Still no joy. I've also cleared cache, logged out and in and reinstalled the app (as suggested in other threads). 

Thank you for prioritizing this. Cheers.

**Reply by aconsuegra** [reply]
Oh, that is unfortunate. Could you please check which version of the app you are using? The fix should be included in 1.137.2

**Reply by jmeridth** [reply]
@aconsuegra 1.138.0 (I'm a beta user)

![image](https://github.com/community/community/assets/35014/9b840eeb-714c-49c3-947b-8b52310118d0)

**Reply by aconsuegra** [reply]
Ah yes, that makes sense, we included the fix only in the production build. The next beta will include it as well!
